### PR TITLE
Purge instance / binding: new flag added "purge" for force & cascade

### PR DESCRIPTION
**Pull request by EpsilonTal**
**Adding a new --force-delete parameter for the deprovision and unbind functions.**
force usage: Use this parameter to delete a resource without raising a confirmation message.
force delete usage: Delete the service instance and all of its associated resources from the database. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access.

### CODE REVIEW

File: docs/commands/deprovision.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Deletes a service instance.
 |--------|-----------|
 | -h, --help  Help for deprovision command.| No |
 | -f, --force Force delete - without confirmation | No |
+| --purge Delete this resource and all its related data | No |
```
Comment (eyalnathan) [inline docs/commands/deprovision.md:19]: how is that different than deleting the resource without `--purge`?

File: docs/commands/unbind.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ Deletes a service binding.
 |--------|-----------|
 | -h, --help  Help for unbind command.| No |
 | -f, --force Force delete without confirmation | No |
+| --purge Delete this resource and all its related data | No |
```
Comment (eyalnathan) [inline docs/commands/unbind.md:19]: same here..

File: internal/cmd/instance/deprovision_test.go [hunk 3]
```diff
@@ -91,6 +93,24 @@ var _ = Describe("Deprovision command test", func() {
 		})
 	})
 
+	Context("when force-delete parameter flag is used", func() {
```
Comment (eyalnathan) [inline internal/cmd/instance/deprovision_test.go:96]:
```suggestion
	Context("when purge flag is used", func() {
```

File: docs/commands/deprovision.md (hunk 1 above)
Comment (EpsilonTal) [reply docs/commands/deprovision.md]: Changed to --force-delete with description by Daniel Korac

File: docs/commands/unbind.md (hunk 2 above)
Comment (EpsilonTal) [reply docs/commands/unbind.md]: Changed to --force-delete with description by Daniel Korac

**Comment by EpsilonTal** [comment]
Adding @idobergerfroind, the open issue of the UA is done now - should be ready for the release.
@eyalnathan reviewed it before, notified again about the UA change.

File: docs/commands/unbind.md [hunk 4]
```diff
@@ -14,13 +14,14 @@ Deletes a service binding.
...
-| -h, --help  Help for unbind command.| No |
-| -f, --force Force delete without confirmation | No |
-| --id ID of the service binding. Required when name is ambiguous| No |
-| --mode  How calls to Service Manager are performed sync or async (default "async")| No |
-| -o, --output Output format of the command. Possible opitons: json, yaml, text.| No|
-| --config Set the path for the smctl config.json file (default is $HOME/.sm/config.json).|Yes|
-| -v, --verbose Use verbose mode.|Yes|
+| -h, --help  Help for unbind command. | No |
+| -f, --force Use this parameter to delete a resource without raising a confirmation message. | No |
+| --force-delete Delete the service instance and all of its associated resources from the database. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access. | No |
```
Comment (eyalnathan) [inline docs/commands/unbind.md:19]: wrong description? this is unbind so it cannot delete "the service instance and all of its..."

File: internal/cmd/binding/unbind.go [hunk 5]
```diff
@@ -53,8 +55,11 @@ func (ubc *UnbindCmd) Prepare(prepare cmd.PrepareFunc) *cobra.Command {
 		RunE:    cmd.RunE(ubc),
 	}
 
-	result.Flags().StringVarP(&ubc.bindingID, "id", "", "", "ID of the service binding. Required when name is ambiguous")
-	result.Flags().BoolVarP(&ubc.force, "force", "f", false, "Force delete without confirmation")
+	forceUsage := "Use this parameter to delete a resource without raising a confirmation message."
+	forceDeleteUsage := "Delete the service instance and all of its associated resources from the database. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access."
```
Comment (eyalnathan) [inline internal/cmd/binding/unbind.go:59]: same here (wrong description)

File: docs/commands/deprovision.md [hunk 6]
```diff
@@ -14,12 +14,13 @@ Deletes a service instance.
...
 |--------|-----------|
-| -h, --help  Help for deprovision command.| No |
-| -f, --force Force delete - without confirmation | No |
-| --id ID of the service instance. Required when name is ambiguous | No |
-| --mode How calls to Service Manager are performed sync or async (default "async") | No |
-| --config Set the path for the smctl config.json file (default is $HOME/.sm/config.json).|Yes|
-| -v, --verbose Use verbose mode.|Yes|
+| -h, --help Help for deprovision command.| No |
+| -f, --force Use this parameter to delete a resource without raising a confirmation message. | No |
+| --force-delete Delete the service instance and all of its associated resources from the database. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access. | No |
```
Comment (eyalnathan) [inline docs/commands/deprovision.md:19]:
```suggestion
| --force-delete Delete the service instance and all of its associated resources from the database, including all service bindings. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access. | No |
```

File: docs/commands/deprovision.md [hunk 7]
```diff
@@ -14,12 +14,13 @@ Deletes a service instance.
...
 |--------|-----------|
-| -h, --help  Help for deprovision command.| No |
-| -f, --force Force delete - without confirmation | No |
-| --id ID of the service instance. Required when name is ambiguous | No |
-| --mode How calls to Service Manager are performed sync or async (default "async") | No |
-| --config Set the path for the smctl config.json file (default is $HOME/.sm/config.json).|Yes|
-| -v, --verbose Use verbose mode.|Yes|
+| -h, --help Help for deprovision command.| No |
+| -f, --force Use this parameter to delete a resource without raising a confirmation message. | No |
+| --force-delete Delete the service instance and all of its associated resources from the database, including all service bindings. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access. | No |
```
Comment (i522006) [inline docs/commands/deprovision.md:19]: Delete the service instance and all of its associated resources from the database, including all service bindings. Use this parameter if the service instance cannot be properly deleted. This parameter can only be used by operators with technical access.

File: docs/commands/deprovision.md [hunk 8]
```diff
@@ -14,12 +14,13 @@ Deletes a service instance.
...
+| --id ID of the service instance. Required when name is ambiguous. | No |
+| --mode How calls to Service Manager are performed sync or async (default "async"). | No |
```
Comment (i522006) [inline docs/commands/deprovision.md:21]: Whether to use synchronous or asynchronous calls to Service Management.
The default value is 'async'.

File: docs/commands/deprovision.md [hunk 9]
```diff
@@ -14,12 +14,13 @@ Deletes a service instance.
...
+| --config Set the path for the smctl config.json file (default is $HOME/.sm/config.json). | Yes |
+| -v, --verbose Use verbose mode. | Yes |
```
Comment (i522006) [inline docs/commands/deprovision.md:23]: Use the Verbose mode.

File: docs/commands/unbind.md [hunk 10]
```diff
@@ -14,13 +14,14 @@ Deletes a service binding.
...
-| -h, --help  Help for unbind command.| No |
-| -f, --force Force delete without confirmation | No |
-| --id ID of the service binding. Required when name is ambiguous| No |
-| --mode  How calls to Service Manager are performed sync or async (default "async")| No |
-| -o, --output Output format of the command. Possible opitons: json, yaml, text.| No|
-| --config Set the path for the smctl config.json file (default is $HOME/.sm/config.json).|Yes|
-| -v, --verbose Use verbose mode.|Yes|
+| -h, --help  Help for unbind command. | No |
+| -f, --force Use this parameter to delete a resource without raising a confirmation message. | No |
+| --force-delete Delete the service binding and all of its associated resources from the database. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access. | No |
```
Comment (i522006) [inline docs/commands/unbind.md:19]: Delete the service instance and all of its associated resources from the database, including all service bindings. Use this parameter if the service instance cannot be properly deleted. This parameter can only be used by operators with technical access.

File: docs/commands/unbind.md [hunk 11]
```diff
@@ -14,13 +14,14 @@ Deletes a service binding.
...
+| --id ID of the service binding. Required when name is ambiguous. | No |
+| --mode  How calls to Service Manager are performed sync or async (default "async"). | No |
```
Comment (i522006) [inline docs/commands/unbind.md:21]: Whether to use synchronous or asynchronous calls to Service Management.
The default value is 'async'.

File: docs/commands/unbind.md [hunk 12]
```diff
@@ -14,13 +14,14 @@ Deletes a service binding.
...
+| -o, --output Output format of the command. Possible options: json, yaml, text. | No |
+| --config Set the path for the smctl config.json file (default is $HOME/.sm/config.json). | Yes |
+| -v, --verbose Use verbose mode. | Yes |
```
Comment (i522006) [inline docs/commands/unbind.md:24]: Use the Verbose mode.

File: docs/commands/unbind.md [hunk 13]
```diff
@@ -14,13 +14,14 @@ Deletes a service binding.
...
-| --mode  How calls to Service Manager are performed sync or async (default "async")| No |
-| -o, --output Output format of the command. Possible opitons: json, yaml, text.| No|
-| --config Set the path for the smctl config.json file (default is $HOME/.sm/config.json).|Yes|
-| -v, --verbose Use verbose mode.|Yes|
+| -h, --help  Help for unbind command. | No |
+| -f, --force Use this parameter to delete a resource without raising a confirmation message. | No |
+| --force-delete Delete the service binding and all of its associated resources from the database. Use this parameter if service instance cannot be properly deleted. This parameters can only be used by operators with technical access. | No |
+| --id ID of the service binding. Required when name is ambiguous. | No |
+| --mode  How calls to Service Manager are performed sync or async (default "async"). | No |
+| -o, --output Output format of the command. Possible options: json, yaml, text. | No |
```
Comment (i522006) [inline docs/commands/unbind.md:22]: The output format of the command. Options: json, yaml, text.

**Review by i522006 — CHANGES_REQUESTED** [review]
I've added a few comments.

**Review by i522006 — APPROVED** [review]

File: internal/cmd/instance/deprovision_test.go [hunk 14]
```diff
@@ -91,6 +93,24 @@ var _ = Describe("Deprovision command test", func() {
 		})
 	})
 
+	Context("when forceDelete flag is used", func() {
```
Comment (eyalnathan) [inline internal/cmd/instance/deprovision_test.go:96]: maybe add a test that makes sure that if we don't use the flag the cascade and force params are not passed? (same for unbind)

**Pull request merged**